### ISSUE TITLE
failing test xt/examples-compilation.t doc/Type/Str.pod6

### ISSUE THREAD

**Issue by coke** [perl6/doc#1238]
From 

```
perl6 xt/examples-compilation.t doc/Type/Str.pod6
```

```
# $string.substr-rw(2, 0) = "-Zorro-"; # insert new characters BEFORE the character at index 2
# $string.say;                         # OUTPUT: «az-Zorro-c␤»
# Variable '$string' is not declared. Did you mean 'Stringy'?
not ok 72 - doc/Type/Str.pod6 chunk 72 compiles
```

**Comment by coke** [comment]
Added in 35e69981b2df84e1151ad2f4117bb27df6534ba2 by @tbrowder

**Comment by hankache** [comment]
This was fixed by 4c0eb45
